File: DSA_programs/BubbleSort.c
/******************************************************************************

10. Write a program for implementation of Bubble sort

*******************************************************************************/
#include<stdio.h>

void swap(int* a, int* b)
{
   int temp = *a;
   *a = *b;
   *b = temp;
}

void bubbleSort(int array[], int size)
{
   for(int i=0 ; i<size-1 ; i++)
   {
      for(int j=0 ; j<size-1-i ; j++)
      {
         if(array[j]>array[j+1])
            swap(&array[j], &array[j+1]);
      }
   }
}

int main()
{
   printf("Enter the size of the array: ");
   int size = 0;
   scanf("%d", &size);

   int array[size];

   printf("\nEnter the array elements:-\n");
   for(int i=0 ; i<size ; i++)
   {
      scanf("%d", &array[i]);
   }

   bubbleSort(array, size);

   printf("\nDisplaying sorted array..\n");
   for(int i=0 ; i<size ; i++)
   {
      printf("%d ", array[i]);
   }
}

File: DSA_programs/BubbleSort.java
import java.util.*;
class BubbleSort{   
  public static void bubble(int arr[], int r, int c){
    if(r==0)
      return;
    if(c<r){
      if(arr[c]>arr[c+1]){
        int temp = arr[c];
        arr[c] = arr[c+1];
        arr[c+1] = temp;
      }
      bubble(arr, r,c+1);
    }
    else{
      bubble(arr, r-1,0);
    }
  }

  public static void iterativeBubble(int arr[]){
    for(int i=0; i<arr.length; i++){
      for(int j=0; j<arr.length-i-1; j++){
        if(arr[i]>arr[i+1]){
          int temp = arr[i];
          arr[i] = arr[i+1];
          arr[i+1] = temp;
        }
      }
    }
    /*for(int i=0; i<arr.length; i++){
      System.out.print(arr[i]+" "); 
    }*/
  }  

  public static void main(String...ar){
    int arr[] = {4,3,2,1};
    //System.out.println(bubble(arr, arr.length-1, 0));
    bubble(arr, arr.length-1, 0);
    System.out.println(Arrays.toString(arr));
    iterativeBubble(arr);
    System.out.println(Arrays.toString(arr));
  }
}